imgix/ember-cli-imgix
# feat: support Ember v2, v3, update component API, fastboot support

BREAKING CHANGE: - Ember v1 no longer supported (although it may still work) 
- Component API changed - `imgix-image` has been renamed to `imgix-image-wrapped`, and `imgix-image-element` has been renamed to `imgix-image`. `imgix-image-wrapped` is deprecated and will be remove in the next major version.

## Description

The majority of this PR is merging the excellent work done by @Duder-onomy in his [fork of this project, ember-imgix](https://github.com/Duder-onomy/ember-imgix). His fork has made significant improvements to this repo, including adding support for newer versions of Ember, and adding fastboot support. 

The new version of `imgix-image` is taken from `ember-imgix`. It has the same API as the old `imgix-image-element`, but just has rewritten internals. I have ensured this compatibility by ensuring that all the integration tests written for the old component still pass for this component. 

The old version of `imgix-image` has been renamed to `imgix-image-wrapped` to signify that it wraps the img with a div. This is still using the old, outdated internals. It is probably a good idea to keep this component for the next major version as it will make migration easier for a lot of users, as they just have to rename their usage of the component, rather than completely change their usage and (potentially) significantly change their CSS. It doesn't seem like a good idea to keep it around permanently as it unnecessarily wraps the img in a `div`, which the user can do themselves if they would like.

After we've discussed and reviewed this PR I'll update the documentation with these changes.

## New Feature

- [x] If this is a big feature with breaking changes, consider [opening an issue][issues] to discuss first. This is completely up to you, but please keep in mind that your PR might not be accepted.

## TODO

- [x] Add deprecation to `imgix-image-wrapped`

## Steps to Test

Steps:

1.  Ensure the ember cli is installed globally `npm i -g ember-cli`
1.  Create a new ember application using `mkdir ember-temp && cd ember-temp && ember init`
1.  Run `git checkout fred/merge-ember-imgix && npm i && npm link` in the ember-cli-imgix repo
1.  Run `ember install ember-cli-imgix && npm link ember-cli-imgix` in the new ember application
1.  Add the following to the APP section of `config/environment.js` (around line 20)
    ```js
    imgix: {
      source: "assets.imgix.net",
      debug: true
    }
    ```
1.  Replace application.hbs with this to show that it's working
    ```hbs
    <div style="width: 100px; height: 100px;">
      {{imgix-image-wrapped path="/examples/pione.jpg"}}
    </div>
    <div style="width: 100px; height: 100px;">
      {{imgix-image path="/examples/pione.jpg"}}
    </div>
    ```
1.  Serve the web application with `ember serve`
